.github/workflows/screener-build.yml

```diff
@@ -75,19 +84,34 @@ jobs:
 
   screener-react-northstar:
     runs-on: 'ubuntu-latest'
-
     name: Screener @fluentui/react-northstar
-
     steps:
       - uses: actions/checkout@v3
         with:
           fetch-depth: 0
 
-      - name: Log environment variables (Linux)
-        if: runner.os == 'Linux'
+      - name: Set env variables if there is a PR
         run: |
-          printenv | sort ;\
-          echo "SHELLOPTS $SHELLOPTS" ;\
+          echo "SYSTEM_PULLREQUEST_TARGETBRANCH=${{ github.event.pull_request.base.ref }}" >> $GITHUB_ENV
+          echo "SYSTEM_PULLREQUEST_SOURCEBRANCH=${{ github.event.pull_request.head.ref }}" >> $GITHUB_ENV
+          echo "SYSTEM_PULLREQUEST_PULLREQUESTID=${{ github.event.pull_request.id }}" >> $GITHUB_ENV
+          echo "SYSTEM_PULLREQUEST_SOURCECOMMITID=${{ github.event.pull_request.head.sha }}" >> $GITHUB_ENV
+          echo "ISPR=true" >> $GITHUB_ENV
+          echo "DEPLOYBASEPATH=pull/$PR_NUMBER" >> $GITHUB_ENV
+          echo "DEPLOYURL=https://$DEPLOYHOST/pull/$PR_NUMBER" >> $GITHUB_ENV
+          echo "BUILD_SOURCEBRANCHNAME='merge'" >> $GITHUB_ENV
+        env:
+          DEPLOYHOST: 'fluentuipr.z22.web.core.windows.net'
+          PR_NUMBER: ${{ github.event.number }}
+        if: ${{startsWith(github.ref, 'refs/pull/')}}
+
+      - name: Set env variables if there is not a PR
+        run: |
+          echo "ISPR=false" >> $GITHUB_ENV
+          echo "DEPLOYBASEPATH=heads/$SYSTEM_PULLREQUEST_TARGETBRANCH" >> $GITHUB_ENV
+          echo "DEPLOYURL=https://$DEPLOYHOST/heads/$SYSTEM_PULLREQUEST_TARGETBRANCH" >> $GITHUB_ENV
+          echo "BUILD_SOURCEBRANCHNAME=${{ github.ref_name }}" >> $GITHUB_ENV
+        if: ${{!startsWith(github.ref, 'refs/pull/')}}
```

Comment:
Blocks with defining variables are three times. Can we put them into a [reusable workflow](https://docs.github.com/en/actions/using-workflows/reusing-workflows)?

Reply:
Now that there is no environment variables duplicate code after downloading them as artifacts for each job, is a reusable workflow necessary?

Reply:
Probably no, let's leave it as it is for now.